### ISSUE TITLE
Sub Downgrade after Switching to a Lower Tier Sub

### ISSUE THREAD

**Issue by c33tz** [Regalis11/Barotrauma#10033]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

TLDNR: basically can't switch between sub tiers without losing all upgrades purchased beyond the lowest Tier sub's limits!! (also appears upgrades transfer from sub to sub)

I had bought a Tier 3 sub, up from my Tier 1, to find i was too broke to properly outfit it after a pretty rough mission, but did upgrade a few systems for the sub. (Notices Upgrades from the Tier 1 sub [Camel] carried over to the new one as well [Winterhalter]

So I switched back to my Tier 1 Sub (also to make more money on transport missions I had) and it appears the sub-upgrades are carrying forward (and back) when switching subs.

I don't think this was intentional that upgrades transfer in the first place, but moreo when I went back to my Tier 1 sub, **then to the Tier 3 again**, I noticed all upgrades that were "out of reach" for the tier one sub were lost and required repurchasing... so effectively the money spend to upgrade the Tier 3 sub, and said upgrades were all lost and reverted back to that of what the Tier 1 sub was upgraded to.

So basically I can't switch between sub tiers without losing all upgrades purchased beyond the Tier 1 limits!!

### Reproduction steps

1. Start up Singleplayer Campaign
2. Visit a Colony
3. Purchase a Higher Tier Sub from Colony than what is on-hand
4. Switch to the Higher Tier sub
5. Purchase Sub Upgrades at the Upgrade merchant
6. Switch back to Lower Tier Sub
7. Switch back to Higher Tier Sub
8. Cry over lost marks & upgrades

### Bug prevalence

Happens regularly

### Version

0.19.11.0

### Which operating system did you encounter this bug on?

Windows

### Relevant error messages and crash reports

```shell
N/A. Game functions "properly", but merely overwriting game/sub data when sapping subs between tiers (with different upgrade limits)
```

**Comment by c33tz** [comment]
Sorry, duplicate bug post afterall -- disregard!
